src/evict/evict_page.c

```diff
@@ -179,9 +179,14 @@ __evict_delete_ref(WT_SESSION_IMPL *session, WT_REF *ref, bool closing)
 		 * something is busy, be sure that the page still ends up
 		 * marked deleted.
 		 */
-		if (ndeleted > pindex->entries / 10 && pindex->entries > 1 &&
-		    (ret = __wt_split_reverse(session, ref)) != EBUSY)
-			return (ret);
+		if (ndeleted > pindex->entries / 10 && pindex->entries > 1) {
+			if ((ret = __wt_split_reverse(session, ref)) == 0)
+				return (ret);
+			WT_RET_BUSY_OK(ret);
+
+			/* If a reverse split fails, unlock the child. */
+			WT_ASSERT(session, ref->state == WT_REF_LOCKED);
```

Comment:
It looks to me like we call `__wt_split_reverse` and immediately return if that's successful, then return if ret is anything other than EBUSY, then fall through to reset the lock. Maybe I'm misreading something, but I think we reset the lock if we get EBUSY here. Let's talk this over tonight -- I can change the `return (ret)` to `return (0)` for clarity, too.